
Revert Vulkan surface update


diff -uarp a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
--- a/dlls/win32u/vulkan.c	2025-11-17 16:03:43.095165417 -0500
+++ b/dlls/win32u/vulkan.c	2025-11-17 16:14:23.616853892 -0500
@@ -173,12 +173,6 @@ static VkSurfaceKHR win32u_wine_get_host
     return surface->host_surface;
 }
 
-static void win32u_vulkan_surface_update( VkSurfaceKHR handle )
-{
-    struct surface *surface = surface_from_handle( handle );
-    driver_funcs->p_vulkan_surface_update( surface->hwnd, surface->driver_private );
-}
-
 static struct vulkan_funcs vulkan_funcs =
 {
     .p_vkCreateWin32SurfaceKHR = win32u_vkCreateWin32SurfaceKHR,
@@ -187,7 +181,6 @@ static struct vulkan_funcs vulkan_funcs
     .p_vkGetDeviceProcAddr = win32u_vkGetDeviceProcAddr,
     .p_vkGetInstanceProcAddr = win32u_vkGetInstanceProcAddr,
     .p_wine_get_host_surface = win32u_wine_get_host_surface,
-    .p_vulkan_surface_update = win32u_vulkan_surface_update,
 };
 
 static VkResult nulldrv_vulkan_surface_create( HWND hwnd, VkInstance instance, VkSurfaceKHR *surface, void **private )
@@ -204,10 +197,6 @@ static void nulldrv_vulkan_surface_detac
 {
 }
 
-static void nulldrv_vulkan_surface_update( HWND hwnd, void *private )
-{
-}
-
 static void nulldrv_vulkan_surface_presented( HWND hwnd, VkResult result )
 {
 }
@@ -227,7 +216,6 @@ static const struct vulkan_driver_funcs
     .p_vulkan_surface_create = nulldrv_vulkan_surface_create,
     .p_vulkan_surface_destroy = nulldrv_vulkan_surface_destroy,
     .p_vulkan_surface_detach = nulldrv_vulkan_surface_detach,
-    .p_vulkan_surface_update = nulldrv_vulkan_surface_update,
     .p_vulkan_surface_presented = nulldrv_vulkan_surface_presented,
     .p_vkGetPhysicalDeviceWin32PresentationSupportKHR = nulldrv_vkGetPhysicalDeviceWin32PresentationSupportKHR,
     .p_get_host_surface_extension = nulldrv_get_host_surface_extension,
@@ -277,12 +265,6 @@ static void lazydrv_vulkan_surface_detac
     return driver_funcs->p_vulkan_surface_detach( hwnd, private );
 }
 
-static void lazydrv_vulkan_surface_update( HWND hwnd, void *private )
-{
-    vulkan_driver_load();
-    return driver_funcs->p_vulkan_surface_update( hwnd, private );
-}
-
 static void lazydrv_vulkan_surface_presented( HWND hwnd, VkResult result )
 {
     vulkan_driver_load();
@@ -306,7 +288,6 @@ static const struct vulkan_driver_funcs
     .p_vulkan_surface_create = lazydrv_vulkan_surface_create,
     .p_vulkan_surface_destroy = lazydrv_vulkan_surface_destroy,
     .p_vulkan_surface_detach = lazydrv_vulkan_surface_detach,
-    .p_vulkan_surface_update = lazydrv_vulkan_surface_update,
     .p_vulkan_surface_presented = lazydrv_vulkan_surface_presented,
 };
 
diff -uarp a/dlls/winemac.drv/vulkan.c b/dlls/winemac.drv/vulkan.c
--- a/dlls/winemac.drv/vulkan.c	2025-11-17 16:00:50.205287426 -0500
+++ b/dlls/winemac.drv/vulkan.c	2025-11-17 16:14:36.385880088 -0500
@@ -182,10 +182,6 @@ static void macdrv_vulkan_surface_detach
 {
 }
 
-static void macdrv_vulkan_surface_update(HWND hwnd, void *private)
-{
-}
-
 static void macdrv_vulkan_surface_presented(HWND hwnd, VkResult result)
 {
 }
@@ -208,7 +204,6 @@ static const struct vulkan_driver_funcs
     .p_vulkan_surface_create = macdrv_vulkan_surface_create,
     .p_vulkan_surface_destroy = macdrv_vulkan_surface_destroy,
     .p_vulkan_surface_detach = macdrv_vulkan_surface_detach,
-    .p_vulkan_surface_update = macdrv_vulkan_surface_update,
     .p_vulkan_surface_presented = macdrv_vulkan_surface_presented,
 
     .p_vkGetPhysicalDeviceWin32PresentationSupportKHR = macdrv_vkGetPhysicalDeviceWin32PresentationSupportKHR,
diff -uarp a/dlls/winevulkan/vulkan.c b/dlls/winevulkan/vulkan.c
--- a/dlls/winevulkan/vulkan.c	2024-11-22 17:08:32.000000000 -0500
+++ b/dlls/winevulkan/vulkan.c	2025-11-17 16:14:47.099861796 -0500
@@ -1786,9 +1786,6 @@ VkResult wine_vkCreateSwapchainKHR(VkDev
     if (surface) create_info_host.surface = surface->host_surface;
     if (old_swapchain) create_info_host.oldSwapchain = old_swapchain->host_swapchain;
 
-    /* update the host surface to commit any pending size change */
-    vk_funcs->p_vulkan_surface_update( surface->driver_surface );
-
     /* Windows allows client rect to be empty, but host Vulkan often doesn't, adjust extents back to the host capabilities */
     res = instance->funcs.p_vkGetPhysicalDeviceSurfaceCapabilitiesKHR(physical_device->host_physical_device,
                                                                       surface->host_surface, &capabilities);
diff -uarp a/dlls/winewayland.drv/vulkan.c b/dlls/winewayland.drv/vulkan.c
--- a/dlls/winewayland.drv/vulkan.c	2025-11-17 16:00:50.205455229 -0500
+++ b/dlls/winewayland.drv/vulkan.c	2025-11-17 16:14:55.383790931 -0500
@@ -118,10 +118,6 @@ static void wayland_vulkan_surface_detac
 {
 }
 
-static void wayland_vulkan_surface_update(HWND hwnd, void *private)
-{
-}
-
 static void wayland_vulkan_surface_presented(HWND hwnd, VkResult result)
 {
     HWND toplevel = NtUserGetAncestor(hwnd, GA_ROOT);
@@ -147,7 +143,6 @@ static const struct vulkan_driver_funcs
     .p_vulkan_surface_create = wayland_vulkan_surface_create,
     .p_vulkan_surface_destroy = wayland_vulkan_surface_destroy,
     .p_vulkan_surface_detach = wayland_vulkan_surface_detach,
-    .p_vulkan_surface_update = wayland_vulkan_surface_update,
     .p_vulkan_surface_presented = wayland_vulkan_surface_presented,
 
     .p_vkGetPhysicalDeviceWin32PresentationSupportKHR = wayland_vkGetPhysicalDeviceWin32PresentationSupportKHR,
diff -uarp a/dlls/winex11.drv/vulkan.c b/dlls/winex11.drv/vulkan.c
--- a/dlls/winex11.drv/vulkan.c	2025-11-17 16:10:26.640144603 -0500
+++ b/dlls/winex11.drv/vulkan.c	2025-11-17 16:15:08.552825648 -0500
@@ -117,13 +117,6 @@ static void X11DRV_vulkan_surface_detach
     }
 }
 
-static void X11DRV_vulkan_surface_update( HWND hwnd, void *private )
-{
-    struct vulkan_surface *surface = private;
-
-    TRACE( "%p %p\n", hwnd, private );
-}
-
 static void X11DRV_vulkan_surface_presented( HWND hwnd, VkResult result )
 {
 }
@@ -147,7 +140,6 @@ static const struct vulkan_driver_funcs
     .p_vulkan_surface_create = X11DRV_vulkan_surface_create,
     .p_vulkan_surface_destroy = X11DRV_vulkan_surface_destroy,
     .p_vulkan_surface_detach = X11DRV_vulkan_surface_detach,
-    .p_vulkan_surface_update = X11DRV_vulkan_surface_update,
     .p_vulkan_surface_presented = X11DRV_vulkan_surface_presented,
 
     .p_vkGetPhysicalDeviceWin32PresentationSupportKHR = X11DRV_vkGetPhysicalDeviceWin32PresentationSupportKHR,
diff -uarp a/include/wine/vulkan_driver.h b/include/wine/vulkan_driver.h
--- a/include/wine/vulkan_driver.h	2025-11-17 16:00:50.205859549 -0500
+++ b/include/wine/vulkan_driver.h	2025-11-17 16:15:23.343215099 -0500
@@ -39,7 +39,6 @@ struct vulkan_funcs
     /* winevulkan specific functions */
     const char *(*p_get_host_surface_extension)(void);
     VkSurfaceKHR (*p_wine_get_host_surface)(VkSurfaceKHR);
-    void (*p_vulkan_surface_update)(VkSurfaceKHR);
 };
 
 /* interface between win32u and the user drivers */
@@ -48,7 +47,6 @@ struct vulkan_driver_funcs
     VkResult (*p_vulkan_surface_create)(HWND, VkInstance, VkSurfaceKHR *, void **);
     void (*p_vulkan_surface_destroy)(HWND, void *);
     void (*p_vulkan_surface_detach)(HWND, void *);
-    void (*p_vulkan_surface_update)(HWND, void *);
     void (*p_vulkan_surface_presented)(HWND, VkResult);
 
     VkBool32 (*p_vkGetPhysicalDeviceWin32PresentationSupportKHR)(VkPhysicalDevice, uint32_t);
-- 
2.47.1

